CMakeLists.txt

```diff
@@ -20,7 +20,7 @@ mxnet_option(USE_F16C             "Build with x86 F16C instruction support" ON)
 mxnet_option(USE_LAPACK           "Build with lapack support" ON)
 mxnet_option(USE_MKL_IF_AVAILABLE "Use MKL if found" ON)
 mxnet_option(USE_MKLML_MKL        "Use MKLDNN variant of MKL (if MKL found)" ON IF USE_MKL_IF_AVAILABLE AND (NOT APPLE))
-mxnet_option(USE_MKLDNN           "Use MKLDNN variant of MKL (if MKL found)" ON IF USE_MKL_IF_AVAILABLE AND (NOT APPLE))
+mxnet_option(USE_MKLDNN           "Use MKLDNN variant of MKL (if MKL found)" ON IF USE_MKL_IF_AVAILABLE AND (NOT APPLE) AND (NOT MSVC) AND (CMAKE_SYSTEM_PROCESSOR MATCHES x86_64))
```

Comment:
Actually `CMAKE_SYSTEM_PROCESSOR` will not work for cross compilation. You could reuse a variable `CMAKE_CROSSCOMPILING` for this as shown [here](https://github.com/apache/incubator-mxnet/pull/13576/files#diff-af3b638bc2a3e6c650974192a53c7291R6).

Reply:
Can you please e

Reply:
@lebeg I checked, `CMAKE_SYSTEM_PROCESSOR` works for cross compilation. `CMAKE_CROSSCOMPILING=TRUE` only for ARM v6, v7, v8, and for all these cases `CMAKE_SYSTEM_PROCESSOR` exists and `CMAKE_SYSTEM_PROCESSOR MATCHES x86_64` returns `FALSE`. Hence, I don't think any change is needed. Please correct me if I am missing anything.

Reply:
Note the semantics of `mxnet_option`. If the condition is not satisfied, this option will be turned off ,not setting the default value off.